### PR TITLE
cert-manager-setup: use crd pattern

### PR DESCRIPTION
**Pull request by s12chung**

### CODE REVIEW

File: staging/cert-manager-setup/files/crds.yaml [hunk 1]
```diff
@@ -7512,4 +7511,3 @@ status:
   conditions: []
```
Comment (jieyu) [inline staging/cert-manager-setup/files/crds.yaml]: Can we remove `"helm.sh/hook": crd-install` annotation?

Comment (jieyu) [reply staging/cert-manager-setup/files/crds.yaml]: oh, i hit approval accidentally

File: staging/cert-manager-setup/templates/install-crds.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      restartPolicy: OnFailure
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cert-manager-setup-crds
+  annotations:
+    "helm.sh/hook": pre-install, pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (jieyu) [inline staging/cert-manager-setup/templates/install-crds.yaml]: Can you also add "before-hook-creation" just to be safe?

"helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation

Comment (jieyu) [reply staging/cert-manager-setup/templates/install-crds.yaml]: Please apply this also to other places (even in kommander chart)

File: staging/cert-manager-setup/files/crds.yaml (hunk 1 above)
Comment (jimmidyson) [reply staging/cert-manager-setup/files/crds.yaml]: No the crd-install hook still needs to be used here.

Comment (s12chung) [reply staging/cert-manager-setup/files/crds.yaml]: yeah, still needs to be here.

File: staging/cert-manager-setup/templates/install-crds.yaml (hunk 2 above)
Comment (s12chung) [reply staging/cert-manager-setup/templates/install-crds.yaml]: done for this chart, there's many places where this needs to be applied. will do it in another PR

File: staging/cert-manager-setup/files/crds.yaml (hunk 1 above)
Comment (jieyu) [reply staging/cert-manager-setup/files/crds.yaml]: Hum? Why?

Comment (s12chung) [reply staging/cert-manager-setup/files/crds.yaml]: i believe helm 2 uses it to check whether it's a CRD or not for validations.

File: staging/cert-manager-setup/templates/install-crds.yaml (hunk 2 above)
Comment (s12chung) [reply staging/cert-manager-setup/templates/install-crds.yaml]: other PR here: https://github.com/mesosphere/charts/pull/242

File: staging/cert-manager-setup/files/crds.yaml (hunk 1 above)
Comment (jieyu) [reply staging/cert-manager-setup/files/crds.yaml]: oh, right. OK.

**Pull request merged**